File: solutions/JavaScript/338.CountingBits.js
var countBits = function (n) {
    let ans = new Array(n + 1).fill(0);
    for (let i = 1; i <= n; i++) {
        ans[i] = countOnes(i);
    }
    return ans;
};

function countOnes(n) {
    let count = 0;
    while (n > 0) {
        if (n % 2 === 1) {
            count++;
        }
        n = Math.floor(n / 2);
    }
    return count;
}

File: solutions/Python/338.CountingBits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        ans = [0] * (n + 1)
        for i in range(1, n + 1):
            ans[i] = self.countOnes(i)
        return ans

    def countOnes(self, n: int) -> int:
        count: int = 0
        while n > 0:
            if n % 2 == 1:
                count += 1
            n //= 2
        return count
